Build failure: pkgsLLVM.zip

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.zip`

### Build log

```
In file included from util.c:28:
./timezone.c:752:15: error: incomplete definition of type 'struct tm'
  752 |   seconds = tm->tm_sec;                 /* 0..61 in ANSI C. */
      |             ~~^
/nix/store/gx8d8b0gamh2ybhgcbf7jmbqqgxcdh68-glibc-aarch64-unknown-linux-gnu-2.39-31-dev/include/wchar.h:94:8: note: forward declaration of 'struct tm'
   94 | struct tm;
      |        ^
In file included from util.c:28:
./timezone.c:776:5: error: incomplete definition of type 'struct tm'
  776 |   tm->tm_year = years - TM_YEAR_BASE;
      |   ~~^
/nix/store/gx8d8b0gamh2ybhgcbf7jmbqqgxcdh68-glibc-aarch64-unknown-linux-gnu-2.39-31-dev/include/wchar.h:94:8: note: forward declaration of 'struct tm'
   94 | struct tm;
      |        ^
In file included from util.c:28:
./timezone.c:777:5: error: incomplete definition of type 'struct tm'
  777 |   tm->tm_mon = months;
      |   ~~^
/nix/store/gx8d8b0gamh2ybhgcbf7jmbqqgxcdh68-glibc-aarch64-unknown-linux-gnu-2.39-31-dev/include/wchar.h:94:8: note: forward declaration of 'struct tm'
   94 | struct tm;
      |        ^
In file included from util.c:28:
./timezone.c:778:5: error: incomplete definition of type 'struct tm'
  778 |   tm->tm_mday = days + 1;
      |   ~~^
/nix/store/gx8d8b0gamh2ybhgcbf7jmbqqgxcdh68-glibc-aarch64-unknown-linux-gnu-2.39-31-dev/include/wchar.h:94:8: note: forward declaration of 'struct tm'
   94 | struct tm;
      |        ^
In file included from util.c:28:
./timezone.c:779:5: error: incomplete definition of type 'struct tm'
  779 |   tm->tm_hour = hours;
      |   ~~^
/nix/store/gx8d8b0gamh2ybhgcbf7jmbqqgxcdh68-glibc-aarch64-unknown-linux-gnu-2.39-31-dev/include/wchar.h:94:8: note: forward declaration of 'struct tm'
   94 | struct tm;
      |        ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
2 warnings and 20 errors generated.
make[1]: *** [unix/Makefile:86: util.o] Error 1
make[1]: Leaving directory '/build/zip30'
make: *** [unix/Makefile:202: generic] Error 2
```

### Additional context

On `aarch64-linux`, nixpkgs master cherry-picked 6234b3e818ccbcf22fafeef17584e88f8298c4e3

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@jktjkt

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.156, NixOS, 24.05 (Uakari), 24.05.20240418.bc279bb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/archivers/zip/default.nix
+++ b/pkgs/tools/archivers/zip/default.nix
@@ -36,6 +36,8 @@ stdenv.mkDerivation rec {
     # Buffer overflow on Unicode characters in path names
     # https://bugzilla.redhat.com/show_bug.cgi?id=2165653
     ./buffer-overflow-on-utf8-rh-bug-2165653.patch
+    # Fixes forward declaration errors with timezone.c
+    ./fix-time.h-not-included.patch
   ] ++ lib.optionals (enableNLS && !stdenv.isCygwin) [ ./natspec-gentoo.patch.bz2 ];
 
   buildInputs = lib.optional enableNLS libnatspec

--- a/pkgs/tools/archivers/zip/fix-time.h-not-included.patch
+++ b/pkgs/tools/archivers/zip/fix-time.h-not-included.patch
@@ -1,0 +1,10 @@
+--- zip30/timezone.c	2024-05-21 05:14:55.959957517 +0000
++++ zip30/timezone.c	2024-05-21 05:15:14.852006815 +0000
+@@ -41,6 +41,7 @@
+ #include "timezone.h"
+ #include <ctype.h>
+ #include <errno.h>
++#include <time.h>
+
+ #ifdef IZTZ_DEFINESTDGLOBALS
+ long timezone = 0;

